rocketpy/tools.py

```diff
@@ -23,3 +26,978 @@ def __get__(self, instance, owner=None):
+
+    def __call__(self, *args):
+        """Adds support for calling a vector as a function, if its elements are
+        callable.
+
+        Parameters
+        ----------
+        args : arguments
+            Arguments to be passed to the vector elements.
+
+        Returns
+        -------
+        Vector
+            Vector with the return of each element called with the given
+            arguments.
+
+        Examples
+        --------
+        >>> v = Vector([lambda x: x**2, lambda x: x**3, lambda x: x**4])
+        >>> v(2)
+        Vector(4, 8, 16)
+        """
+        try:
+            return self.element_wise(lambda f: f(*args))
+        except TypeError as exc:
+            msg = "One or more elements of this vector is not callable."
+            raise TypeError(msg) from exc
+
+    def __len__(self):
+        return 3
+
+    @cached_property
+    def x(self):
+        """First component of the vector."""
+        return self.components[0]
+
+    @cached_property
+    def y(self):
+        """Second component of the vector."""
+        return self.components[1]
+
+    @cached_property
+    def z(self):
+        """Third component of the vector."""
+        return self.components[2]
+
+    @cached_property
+    def unit_vector(self):
+        """R3 vector with the same direction of self, but normalized."""
+        return self / abs(self)
+
+    @cached_property
+    def cross_matrix(self):
+        """Skew symmetric matrix used for cross product.
+
+        Notes
+        -----
+        The cross product between two vectors can be computed as the matrix
+        product between the cross matrix of the first vector and the second
+        vector.
+
+        Examples
+        --------
+        >>> v = Vector([1, 7, 3])
+        >>> u = Vector([2, 5, 6])
+        >>> (v ^ u) == v.cross_matrix @ u
+        True
+        """
+        return Matrix(
+            [[0, -self.z, self.y], [self.z, 0, -self.x], [-self.y, self.x, 0]]
+        )
+
+    def __abs__(self):
+        """R3 vector norm, magnitude or absolute value."""
+        return (self.x**2 + self.y**2 + self.z**2) ** 0.5
+
+    def __neg__(self):
+        """-1 times R3 vector self."""
+        return Vector([-self.x, -self.y, -self.z])
+
+    def __add__(self, other):
+        """Sum two R3 vectors."""
+        result = [self[i] + other[i] for i in range(3)]
+        return Vector(result)
+
+    def __radd__(self, other):
+        """Sum two R3 vectors."""
+        return self.__add__(other)
+
+    def __sub__(self, other):
+        """Subtract two R3 vectors."""
+        result = [self[i] - other[i] for i in range(3)]
+        return Vector(result)
+
+    def __rsub__(self, other):
+        """Subtract two R3 vectors."""
+        result = [other[i] - self[i] for i in range(3)]
+        return Vector(result)
+
+    def __mul__(self, other):
+        """Component wise multiplication between R3 vector and scalar other."""
+        return self.__rmul__(other)
```

Comment:
Adding to the discussion, I performed some `timeit` tests for an analogous `__mul__` case and have the following relative performance:

```
Mul:     0.5847252989999561
Short RMul:      0.6325959810001223
Mul:     0.5638121239999236
Explicit RMul:   0.5642033120000178
Ratio 1.1212198998932894
```

All in seconds after `1_000_000` runs. The simple operations should be the worst case scenario, since they are the "fastest" operations relative to the extra function call time. Therefore, the worst case is about `10 %` slower.

In my opinion, it is worth making explicit this `__mul__`. I will leave the suggestion, but feel free to decide what you think is best here.

```suggestion
        return Vector([other * self.x, other * self.y, other * self.z])
```